src/main/kotlin/no/nav/tilleggsstonader/sak/infrastruktur/database/AdvisoryLockService.kt

```diff
@@ -0,0 +1,38 @@
+package no.nav.tilleggsstonader.sak.infrastruktur.database
+
+import org.slf4j.LoggerFactory
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+
+/**
+ * Service for å låse eks en behandlingId før man sjekker om noe finnes i databasen for å sen opprette noe koblet til behandlingen.
+ * Row lock, med eks @Lock(LockMode.PESSIMISTIC_WRITE) i Repository fungerer hvis tabellen allerede inneholder en rad
+ */
```

Comment:
Syntes setningen "Service for å låse eks en behandlingId før man sjekker om noe finnes i databasen for å sen opprette noe koblet til behandlingen." var litt vanskelig å forstå.

Klarte ikke helt å skrive den om, men noe ala

"låse en behandlingId, slik at kall som først sjekker om noe finnes i databasen før den opprettet noe koblet til behandlingen blir stoppet dersom det den leter etter opprettes i en annen transaksjon"

Det forklares også litt på selve funksjonen nede så kan hende at det hadde holdt å bare utvide en av de og ikke ha to kommentarer